lib/linguist/languages.yml

```diff
@@ -2090,13 +2090,40 @@ JSON:
   - composer.lock
   - mcmod.info
   language_id: 174
-JSON5:
+JSON with comments:
   type: data
+  group: JSON
+  tm_scope: source.js
+  ace_mode: javascript
+  codemirror_mode: javascript
+  codemirror_mime_type: text/javascript
+  aliases:
+  - jsonc
   extensions:
-  - ".json5"
+  - ".jsonc"
+  - ".sublime-build"
+  - ".sublime-commands"
+  - ".sublime-completions"
+  - ".sublime-keymap"
+  - ".sublime-macro"
+  - ".sublime-menu"
+  - ".sublime-mousemap"
+  - ".sublime-project"
+  - ".sublime-settings"
+  - ".sublime-theme"
+  - ".sublime-workspace"
+  - ".sublime_metrics"
+  - ".sublime_session"
   filenames:
   - ".babelrc"
+  - ".eslintrc.json"
   - ".jslintrc"
+  - "tsconfig.json"
```

Comment:
Double quotes are not necessary here since the filename doesn't start with a dot.